plugins/modules/aci_management_network_instance_profile.py

```diff
@@ -0,0 +1,301 @@
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+from ansible_collections.cisco.aci.plugins.module_utils.constants import VALID_QOS_CLASSES
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        profile=dict(type="str", aliases=["name", "profile_name"]),
+        qos_class=dict(type="str", aliases=["qos", "priority", "prio"], choices=VALID_QOS_CLASSES),
+        description=dict(type="str", aliases=["descr"]),
+        subnets=dict(type="list", elements="str", aliases=["subnet_list", "networks", "network_list"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["profile"]],
+            ["state", "present", ["profile"]],
+        ],
+    )
+
+    profile = module.params.get("profile")
+    qos_class = module.params.get("qos_class")
+    description = module.params.get("description")
+    subnets = module.params.get("subnets")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class="mgmtInstP",
+            aci_rn="tn-mgmt/extmgmt-default/instp-{0}".format(profile),
+            module_object=profile,
+            target_filter={"name": profile},
+        ),
+        child_classes=["mgmtSubnet"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if subnets:
```

Comment:
Shall we give users the ability to remove all existing subnet entries if an empty list is provided under subnets?
```
- name: Remove existing subnets
  cisco.aci.aci_management_network_instance_profile:
    host: apic
    username: admin
    password: SomeSecretPassword
    name: lab_network_inst_profile
    subnets: []
    state: present
  delegate_to: localhost
```
I believe this modification should suffice for that?
`if subnets is not None:`